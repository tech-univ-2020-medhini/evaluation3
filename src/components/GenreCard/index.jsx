import React from 'react';
import PropTypes from 'prop-types';

import Button from '../Button';


function GenreCard(props) {
  const { name, imageLink, getSongs } = props;
  return (
    <div className="GenreCard">
      <img src={imageLink} alt="/" />
      {/* <Link to="/songs?genre=:name"> */}
      <Button text={name} click={getSongs} />
      {/* </Link> */}
    </div>

  );
}
GenreCard.propTypes = {
  name: PropTypes.string.isRequired,
  imageLink: PropTypes.string.isRequired,
  getSongs: PropTypes.func.isRequired,
};

export default GenreCard;
