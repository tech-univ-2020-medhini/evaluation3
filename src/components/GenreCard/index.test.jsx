import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import GenreCard from './index';


describe('the genre cards', (w) => {
  it('should render correctly', () => {
    const { asFragment } = render(<GenreCard />);

    expect(asFragment()).toMatchSnapshot();
  });
  it('should call get songs when the button is clicked', () => {
    const mockGetSongs = jest.fn();
    const { getByTestId } = render(<GenreCard getSongs={mockGetSongs} />);
    fireEvent.click(getByTestId('test-btn'));
    expect(mockGetSongs).toHaveBeenCalled();
  });
});
