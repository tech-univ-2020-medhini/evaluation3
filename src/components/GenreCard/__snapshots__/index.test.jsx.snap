// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the genre cards should render correctly 1`] = `
<DocumentFragment>
  <div
    class="genre-card"
  >
    <img
      alt="/"
    />
    <button
      class="Button"
      data-testid="test-btn"
      type="button"
    />
  </div>
</DocumentFragment>
`;
